Automattic/wp-calypso
# Analytics: rewrite trackAffiliateReferral from data layer to simple fetch

When recording Calypso page views that have `aff=` query params (and a few others like `cid=` or `ref=`), we record a referral by pinging `https://refer.wordpress.com` and recording the response (success or error) in Tracks as `calypso_refer_visit_response` event.

The current implementation uses data layer, which is an overkill for an one-off request like this that's issued only from one place and doesn't need to store the response in state.

This PR rewrites the code to a simple `fetch` call and removes the Redux dependency from `lib/analytics`.

**How to test:**
#27888 has well described testing steps that apply to this PR equally well. Load a Calypso page with valid and invalid `aff=` query params, and verify that the `refer.wordpress.com` request returned an expected response and recorded it in Tracks as expected.
